gateway/mw_go_plugin.go

```diff
@@ -122,6 +125,12 @@ func (m *GoPluginMiddleware) loadPlugin() bool {
 
 	// try to load plugin
 	var err error
+
+	if !FileExist(m.Path) {
+		// if the exact name doesn't exist then try to load it using tyk version
+		m.Path = m.goPluginFromTykVersion()
```

Comment:
@sredxny I think it will be a good idea to check if the new file path exists after `m.goPluginFromTykVersion()`

Reply:
@tbuchaillot the instruction `goplugin.GetHandler `will check that, in case that the file doesnt exist then a log is displayed like this:

`ERROR Could not load Go-plugin error=plugin.Open("mygoplugin/mygoplugin_v4.1.0_darwin_amd64.so"): realpath failed mwPath=./mygoplugin/mygoplugin.so mwSymbolName=AddFooBarHeader`.

In the case of bundles it will not be affected as the process is:
1- You set the name of the bundle in the api definition (it doesn't have to be the same name of the plugin)
2- On load Apis the gateway will fetch the bundle from the file server and store it locally
3- The gateway process the manifest and assign those values to the CustomMiddleware key... check https://github.com/TykTechnologies/tyk/blob/master/gateway/coprocess_bundle.go#L101
4- The plugin is loaded with the logic of `func (m *GoPluginMiddleware) loadPlugin()` so at this point it doesnt really matters if the values of custom middleware are set via bundle or directly into the api definition